calcite-qml/Calcite/RadioDelegate.qml

```diff
@@ -15,34 +16,62 @@
  ******************************************************************************/
 import QtQuick 2.15
 import QtQuick.Templates 2.15 as T
+import QtGraphicalEffects 1.15
 
 T.RadioDelegate {
     id: control
 
-    implicitWidth: Math.max(background ? background.implicitWidth : 0,
-                            contentItem.implicitWidth + leftPadding + rightPadding)
-    implicitHeight: Math.max(background ? background.implicitHeight : 0,
-                             Math.max(contentItem.implicitHeight,
-                                      indicator ? indicator.implicitHeight : 0) + topPadding + bottomPadding)
+    Component.onCompleted: console.log(width)
+    implicitWidth: Math.max(
+                       background ? background.implicitWidth : 0,
+                       contentItem.implicitWidth + leftPadding + rightPadding)
+    implicitHeight: Math.max(
+                        background ? background.implicitHeight : 0, Math.max(
+                            contentItem.implicitHeight,
+                            indicator ? indicator.implicitHeight : 0) + topPadding + bottomPadding)
 
     padding: 5
     spacing: 5
 
+    icon.width: 24
+    icon.height: 24
+    icon.source: "images/bullet-point.svg"
+    font.bold: control.checked || control.highlighted
+
     contentItem: Text {
-        leftPadding: control.mirrored ? control.indicator.width + control.spacing : 0
-        rightPadding: !control.mirrored ? control.indicator.width + control.spacing : 0
+        leftPadding: !control.mirrored ? control.indicator.width + control.spacing : 0
+        rightPadding: control.mirrored ? control.indicator.width + control.spacing : 0
         text: control.text
         font: control.font
         color: control.enabled ? Calcite.text1 : Calcite.text3
         elide: Text.ElideRight
         visible: control.text
-        horizontalAlignment: Text.AlignHCenter
+        horizontalAlignment: Text.AlignLeft
         verticalAlignment: Text.AlignVCenter
     }
 
-    indicator: RadioIndicator {
-        x: control.mirrored ? control.leftPadding : control.width - width - control.rightPadding
+    indicator: Image {
+        id: indicator
+        visible: control.checked || control.highlighted || control.hovered
+        x: control.mirrored ? control.width - width - control.rightPadding : control.leftPadding
         y: control.topPadding + (control.availableHeight - height) / 2
-        control: control
+        source: icon.source
+        sourceSize: Qt.size(icon.width, icon.height)
+        width: icon.width
+        height: icon.height
+        ColorOverlay {
+            anchors.fill: indicator
+            source: indicator
+            color: control.checked
+                   || control.highlighted ? Calcite.brand : control.hovered ? Calcite.border1 : "transparent"
+            visible: indicator.visible
+        }
+    }
```

Comment:
❌ (This is my fault by taking the initial implementation from an older Material implementation.) I'd just like to quickly compare to how Material's RadioIndicator works as of Qt 5.15.7, and how it interacts with `icon`:

* Materials uses an `IconLabel` which takes an icon property. Here we are using   a plain `Text`.
* Material's `indicator` is independent of the icon. Here `icon` and `indicator` are linked.

I believe that you should take the same approach as Material. The `icon` should be an optional item that can be added to the `RadioDelegate` to show an image. If the user wants to change the indicator they should override the `indicator` property, not the `icon` property.

Material implementation below:

```qml
    indicator: RadioIndicator {
        x: control.text ? (control.mirrored ? control.leftPadding : control.width - width - control.rightPadding) : control.leftPadding + (control.availableWidth - width) / 2
        y: control.topPadding + (control.availableHeight - height) / 2
        control: control
    }

    contentItem: IconLabel {
        leftPadding: !control.mirrored ? 0 : control.indicator.width + control.spacing
        rightPadding: control.mirrored ? 0 : control.indicator.width + control.spacing

        spacing: control.spacing
        mirrored: control.mirrored
        display: control.display
        alignment: control.display === IconLabel.IconOnly || control.display === IconLabel.TextUnderIcon ? Qt.AlignCenter : Qt.AlignLeft

        icon: control.icon
        text: control.text
        font: control.font
        color: control.enabled ? control.Material.foreground : control.Material.hintTextColor
    }

    background: Rectangle {
        implicitHeight: control.Material.delegateHeight

        color: control.highlighted ? control.Material.listHighlightColor : "transparent"

        Ripple {
            width: parent.width
            height: parent.height

            clip: visible
            pressed: control.pressed
            anchor: control
            active: control.down || control.visualFocus || control.hovered
            color: control.Material.rippleColor
        }
    }
```

Reply:
Ok, by using an `IconLabel` we can follow even more closely the Calcite standard by having an extra image at the beginning of the component (`icon`). I am using the `indicator` to show the bullet point as default.
![image](https://user-images.githubusercontent.com/91129623/153454887-b180958b-a1ed-4a29-9426-4641223632ea.png)